core-api/src/controllers/auth.ts

```diff
@@ -0,0 +1,47 @@
+import { Request, Response } from "express";
+import { AuthService } from "../services/auth";
+import { HttpStatus } from "../utils/httpStatus";
+import { ServerError, ErrorCode } from "../utils/serverError";
+import Messages from "../utils/messages";
+
+export class AuthController {
+    public static async signUp(req: Request, res: Response) {
+        const { name, email, password } = req.body;
+
+        if (!name || !email || !password) {
```

Comment:
It's a better approach for performance and readability to start using a if (name && email && password) with the newUser and else with the error.